### PR TITLE
Feat/Save form draft data to localStorage

### PR DESCRIPTION
**Pull request by kkatusic**

### CODE REVIEW

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx [hunk 1]
```diff
@@ -33,63 +29,91 @@ const getErrorMessage = (
...
-	const handleButtonClick = () => {
-		if (inputRef.current) {
-			inputRef.current.click();
+	// Load data from localStorage when the component mounts
+	useEffect(() => {
+		try {
+			const storedData = JSON.parse(
+				localStorage.getItem('proposalFormData') || '{}',
+			);
+			setProposalFormData({
```
Comment (b0rza) [inline apps/platform/src/app/[chainId]/projects/propose/page.tsx:73]: How about 
```javascript
if (storedData) setProposalFormData(storedData);
```
The `useState` will make sure that all the props are present and set to `''` on init. No need to 'destructure' and check for defaults.

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx [hunk 2]
```diff
@@ -33,63 +29,91 @@ const getErrorMessage = (
...
-	};
-
-	const handleButtonClick = () => {
-		if (inputRef.current) {
-			inputRef.current.click();
+	// Load data from localStorage when the component mounts
+	useEffect(() => {
+		try {
+			const storedData = JSON.parse(
+				localStorage.getItem('proposalFormData') || '{}',
```
Comment (b0rza) [inline apps/platform/src/app/[chainId]/projects/propose/page.tsx:71]: Extract `'proposalFormData'` to a constant for reference throughout.

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx [hunk 3]
```diff
@@ -33,63 +29,91 @@ const getErrorMessage = (
...
+		email: '',
+		mobile: '',
+		isChecked: false,
+	});
 
 	const handleCheckboxChange = () => {
 		setIsChecked(prevState => !prevState);
+		setProposalFormData(prevData => ({
+			...prevData,
+			isChecked: !prevData.isChecked,
```
Comment (b0rza) [inline apps/platform/src/app/[chainId]/projects/propose/page.tsx:58]: I would move `isChecked` out of saved form data. It's something that has to be explicitly checked, and not a part of their proposal draft.

**Review by b0rza — CHANGES_REQUESTED** [review]
Nice 💪 
minor stuff to clean up...

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx (hunk 3 above)
Comment (kkatusic) [reply apps/platform/src/app/[chainId]/projects/propose/page.tsx:58]: Good suggstion, thx, removed

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx (hunk 2 above)
Comment (kkatusic) [reply apps/platform/src/app/[chainId]/projects/propose/page.tsx:71]: thx, junior mistake :)

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx (hunk 1 above)
Comment (kkatusic) [reply apps/platform/src/app/[chainId]/projects/propose/page.tsx:73]: thx, junior mistake :)

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx (hunk 2 above)
Comment (kkatusic) [reply apps/platform/src/app/[chainId]/projects/propose/page.tsx:71]: done, thx

File: apps/platform/src/app/[chainId]/projects/propose/page.tsx (hunk 1 above)
Comment (kkatusic) [reply apps/platform/src/app/[chainId]/projects/propose/page.tsx:73]: not junior mistake :D, now I remember it is about typescript issue, typescript looks at it like undefined on input default values I refactor code on other way now it's ok ;)

**Review by b0rza — APPROVED** [review]

**Pull request merged**